internal/pkg/template/templates/workloads/partials/cf/rollback-alarms.yml

```diff
@@ -0,0 +1,45 @@
+{{- if .DeploymentConfiguration.CPUUtilization}}
+CPURollbackAlarm:
+  Metadata:
+    'aws:copilot:description': "A CloudWatch alarm associated with CPU utilization for deployment rollbacks"
+  Type: AWS::CloudWatch::Alarm
+  DependsOn: Service
```

Comment:
Is this because of `!Ref Service`?

iirc with the `!Ref Service`, CFN should be able to work out the dependency graph already - i.e. this alarm will be created *after* service. So we probably don't need `DependsOn: Service` in this case!

Reply:
However, this makes me wonder - if the `Service` creation/update is already completed, does that mean the deployment already stabilized before the  creation of the alarm? Should the alarm have been created.updated before the `Service` instead?

Reply:
It's a very funny thing-- the service has to exist before the alarm can be created; but unless we want users to have to deploy once, then add the alarm name, then deploy again, we need to make sure the service is created before the alarm, and assign the Copilot-created name in the `deployments` field. 

The alarm names can exist/ABR can be enabled during service creation, but ABR is only applied during `UpdateService` because you can only roll back to a previous version, so ABR doesn't apply to the first deployment.

EDS (ECS Deployments Service) doesn't validate that the given alarm exists in an account (though I'm not sure how the bake time is calculated in this case), it just tries to poll for a non-existent alarm. So in theory the service can exist before the alarm, but the alarm still can't exist before the service, hence the `DependsOn`. I plan to add a validation of any passed-in `AlarmNames` in the interest of failing faster.

Reply:
Interesting. Just validating my understanding. If user changes `cpu_utilization` as well as some application code, then the `CPUUtilizationAlarm` will actually get updated *after* the service, meaning that the alarm will not be used in this deployment, not until the next deployment. Is that correct?

Reply:
I tested it and it was used in the current deployment. There's enough time (the deployment becomes long with ABR enabled) for the alarm to get created/updated.
"The bake time is a period of time after the new service version has reached steady state, during which Amazon ECS continues to monitor the alarm associated with the deployment." from https://aws.amazon.com/about-aws/whats-new/2022/12/amazon-ecs-cloudwatch-alarms-safety-deployments/ and for more, https://aws.amazon.com/blogs/containers/automate-rollbacks-for-amazon-ecs-rolling-deployments-with-cloudwatch-alarms/

What isn't great is that not a lot of info is surfaced by the progress tracker so it just seems to be taking forever/stalling and when a rollback happens, it's not obvious. I'll address that in a coming PR.

Reply:
Actually, upon further testing, you're right, @Lou1415926. I watched the multiple consoles and the alarm values didn't update until afterward.

Reply:
This is the current state for rollbacks:
ECS Console:
![Screen Shot 2022-12-23 at 9 52 42 AM](https://user-images.githubusercontent.com/60631893/209383839-377b65ad-ce56-4c49-865a-3a4c9ac01ab6.png)

Progress tracker:
<img width="1060" alt="Screen Shot 2022-12-23 at 9 53 19 AM" src="https://user-images.githubusercontent.com/60631893/209383948-0cb2b0a8-e8e3-4cb6-b587-c8f979af604f.png">

Reply:
😭

Reply:
What about without `--force` flag? iirc, with `--force` flag, Copilot just waits for `UpdateService` to return, whereas without `--force` flag Copilot keeps reading information about the service and sort of "judge" whether the deployment is finished from the service information.

Reply:
without the `DependsOn:`, got into a `UPDATE_ROLLBACK_FAILED` state

Reply:
The larger problem I'm having right now. Circuit Breaker seems to be interrupting the rollback. 😭 
<img width="1398" alt="Screen Shot 2022-12-23 at 10 07 10 AM" src="https://user-images.githubusercontent.com/60631893/209400166-16e08ce4-eb36-49fc-990f-fd9bba30092d.png">

Reply:
Update:
1. I'm removing the `DependsOn` because alarms aren't polled when a service is created, but this way they can be polled (if updated) in subsequent deployments
2. EDS is working on changing the Circuit Breaker error message that was being returned even if CB was disabled. 
3. @Lou1415926 was right-- the difference in behavior with and without `--force` is the main issue right now. Without `--force`, CFN gets in an `UPDATE_ROLLBACK_FAILED` every time an alarm is triggered.